gc_manager error format does not meet expectations

## Bug Report

<!-- Thanks for your bug report! Don't worry if you can't fill out all the sections. -->

### What version of TiKV are you using?
<!-- You can run `tikv-server --version` -->
v4.0.2
### What operating system and CPU are you using?
<!-- If you're using Linux, you can run `cat /proc/cpuinfo` -->

### Steps to reproduce
<!-- If possible, provide a recipe for reproducing the error. A complete runnable program is good. -->

Because of the region has been scheduled, the GC delete region happened error and try again later. 

### What did you expect?
```log
[2020/07/16 15:12:46.102 +08:00] [WARNING] [gc_manager.rs:533] ["failed gc"] [err="Engine(Request(message: \"stale command\"))"] [end_key=748000000000003EFFB75F698000000000FF00000D0003800000FF0000015CAB000000FC] [region_epoch="Some(conf_ver: 6761 version: 11644)"] [region_id=15161048] offset:307,243,524 source:/data/tikv/tikvlog/tikv.logtype:log _id:AXNWeAkcWV1P0-Qo9HjW _type:doc _index:logstash-tikv-2020.07.16 _score: -
```
### What did happened?
```log
[2020/07/16 15:12:46.102 +08:00] [ERROR] [gc_manager.rs:533] ["failed gc"] [err="Engine(Request(message: \"stale command\"))"] [end_key=748000000000003EFFB75F698000000000FF00000D0003800000FF0000015CAB000000FC] [region_epoch="Some(conf_ver: 6761 version: 11644)"] [region_id=15161048] offset:307,243,524 source:/data/tikv/tikvlog/tikv.logtype:log _id:AXNWeAkcWV1P0-Qo9HjW _type:doc _index:logstash-tikv-2020.07.16 _score: -
```

--- a/src/server/gc_worker/gc_manager.rs
+++ b/src/server/gc_worker/gc_manager.rs
@@ -546,7 +546,7 @@ impl<S: GcSafePointProvider, R: RegionInfoProvider> GcManager<S, R> {
         ) {
             // Ignore the error and continue, since it's useless to retry this.
             // TODO: Find a better way to handle errors. Maybe we should retry.
-            error!("failed gc"; "start_key" => &hex_start, "end_key" => &hex_end, "err" => ?e);
+            warn!("failed gc"; "start_key" => &hex_start, "end_key" => &hex_end, "err" => ?e);
         }
 
         *processed_regions += 1;

